library/item/item/src/main/java/org/quiltmc/qsl/item/api/item/setting/CustomDamageHandler.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2021-2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.item.api.item.setting;
+
+import java.util.function.Consumer;
+
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.item.ItemStack;
+
+/**
+ * Allows an item to run custom logic when {@link ItemStack#damage(int, LivingEntity, Consumer)} is called.
+ * This is useful for items that, for example, may drain durability from some other source before damaging
+ * the stack itself.
+ *
+ * <p>Custom damage handlers can be set with {@link QuiltItemSettings#customDamage}.
+ */
+@FunctionalInterface
+public interface CustomDamageHandler {
+	/**
+	 * Called to apply damage to the given stack.
+	 * This can be used to e.g. drain from a battery before actually damaging the item.
+	 *
+	 * @param stack The {@link ItemStack} that is being damaged
```

Comment:
PR-wide comment: `@param`/`@return`/`@throws`/etc should not have the first word capitalized (in the conventions).

Reply:
ill fix with my next commit